src/Utilities/VulnerabilityRoutes.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState, lazy, Suspense, Fragment } from 'react';
+import { Route, Routes, useLocation } from 'react-router-dom';
+import { getSystems } from '../Helpers/APIHelper';
+import { PATHS } from '../Helpers/constants';
+import { intl } from './IntlProvider';
+import messages from '../Messages';
+import AsyncComponent from '@redhat-cloud-services/frontend-components/AsyncComponent';
+import ErrorState from '@redhat-cloud-services/frontend-components/ErrorState';
+
+const SystemsPage = lazy(() =>
+    import(
+        /* webpackChunkName: "Systems" */ '../Components/SmartComponents/SystemsPage/SystemsPage'
+    )
+);
+
+const LandingPage = lazy(() =>
+    import(
+        /* webpackChunkName: "LandingPage" */ '../Components/SmartComponents/LandingPage/LandingPage'
+    )
+);
+
+const Upgrade = lazy(() =>
+    import(
+        /* webpackChunkName: "Upgrade" */ '../Components/PresentationalComponents/StaticPages/UpgradePage'
+    )
+);
+
+const CVEDetailsPage = lazy(() =>
+    import(
+        /* webpackChunkName: "CVEDetailsPage" */ '../Components/SmartComponents/CVEDetailsPage/CVEDetailsPage'
+    )
+);
+
+const SystemDetailsPage = lazy(() =>
+    import(
+        /* webpackChunkName: "SystemDetailsPage" */ '../Components/SmartComponents/SystemDetailsPage/SystemDetailsPage'
+    )
+);
+const Reports = lazy(() =>
+    import(/* webpackChunkName: "Reports" */ '../Components/SmartComponents/Reports/ReportsPage')
+);
+
+const buildTitle = (title, location) => {
+    const subPath = location.pathname && location.pathname.split('/')[4];
+    document.title = `${subPath ? `${subPath} - ` : ''} ${title} - ${intl.formatMessage(messages.pageTitleSuffix)}`;
+};
+
+export const VulnerabilityRoutes = () => {
+    let location = useLocation();
+    const [hasSystems, setHasSystems] = useState(true);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            const result = await getSystems();
+            setHasSystems(result?.meta?.total_items > 0);
+        };
+
+        fetchData();
+    }, []);
+
+    return (
+        // TODO: Discuss with UX some nice loading placeholder
```

Comment:
I think we use the `Spinner` component in most cases where we load a page/component.